geomstats/geometry/discrete_surfaces.py

```diff
@@ -931,7 +931,92 @@ def path_energy(self, path):
         """
         return 0.5 * gs.sum(self.path_energy_per_time(path))
 
-    def _ivp(self, initial_point, initial_tangent_vec, times):
+    def exp(self, tangent_vec, base_point):
+        """Compute the exponential map.
+
+        Parameters
+        ----------
+        tangent_vec : array-like, shape=[..., dim]
+            Tangent vector at base point.
+        base_point : array-like, shape=[..., dim]
+            Base point.
+
+        Returns
+        -------
+        exp : array-like, shape=[..., dim]
+            End point of the geodesic starting at base_point with
+            initial velocity tangent_vec and stopping at time 1.
+        """
+        return self.exp_solver.exp(self._space, tangent_vec, base_point)
+
+    def log(self, point, base_point):
+        """Compute the logarithm map.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., dim]
+            Point.
+        base_point : array-like, shape=[..., dim]
+            Base po int.
+
+        Returns
+        -------
+        tangent_vec : array-like, shape=[..., dim]
+            Initial velocity of the geodesic starting at base_point and
+            reaching point at time 1.
+        """
+        return self.log_solver.log(self._space, point, base_point)
+
+
+class _ExpSolver:
+    def __init__(self, n_steps=10, optimizer=None):
+        if optimizer is None:
+            optimizer = ScipyMinimize(
+                method="L-BFGS-B",
+                jac="autodiff",
+                options={"disp": False, "ftol": 0.00001},
+            )
+
+        self.n_steps = n_steps
+        self.optimizer = optimizer
+
+    def exp(self, space, tangent_vec, base_point):
```

Comment:
can we call this `forward` so that we can do `exp_solver()` directly instead of `exp_solver.exp()`?